build.sh

```diff
@@ -16,6 +16,27 @@ TMP="$PWD/tmp"
 DL="$PWD/dl"
 PACKAGE="$PWD/package"
 
+if [[ $OSTYPE == 'darwin'* ]]; then
```

Comment:
This isn't really unique to macOS; these things aren't installed by default on Linux either.

I'm not sure we necessarily care about these kinds of checks in the build script. These dependencies are already documented in the [m1n1 repo](https://github.com/asahilinux/m1n1) README. If anything the README here should reference that.

I don't really expect users to be building the installer themselves, so there's no need for hand-holding here; I expect people hacking on the installer to know how to read a README and install some dependencies ;)

Reply:
I agree, the only thing that might be darwin specific is using clang over gcc, but that also requires they have command line tools or Xcode installed.

You are also correct about the m1n1 README, but you are assuming people have been closely following your work and/or already built it previously using just the m1n1 README.  I got it working myself on a machine ~ a month ago, but then saw the installer and was excited to give it a go and brand new hardware (and had forgot a lot of the steps I had done to get it to work before.)  I immediately hit a bunch of dep issues (the first being the installer expects to be ran in linux where cpio has the -D flag as I thought I'd be able to run on macOS.

I think in this world of single click installs or "curl | bash" installers, there is an assumption of it just working perfectly out of the box.  

I think the only real value perhaps in my PR would be just to point out to you how you can leverage Github Actions to build and host your installer in a transparent way.

Reply:
Well, that's why the installer will be `curl https://alx.sh | sh`... once it's ready for end users :-)

(and for now, it's `curl https://mrcn.st/alxsh | sh`, I'm just not advertising that widely because it's really meant for people who are closely following the project, so I only mentioned it on IRC)

Reply:
Very cool! Sounds like I need to start hanging out in the irc 😎